### PR TITLE
Pandas lesson

### PR DESCRIPTION
**Pull request by derekhoward**
The entire lesson is in the notebook, I will upload a complete/updated version after the session.

### CODE REVIEW

File: lessons/pandas_walkthrough/lesson.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+layout: page
+title: Pandas for data analysis
+visible: false
+---
+<!-- change visible to true if you want it on the site -->
```
Comment (QuLogic) [inline lessons/pandas_walkthrough/lesson.md:6]: I think you might want to read this comment...

File: lessons/pandas_walkthrough/lesson.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ - **Authors**: Derek Howard
+ - **Research field**: Neuroscience
+ - **Lesson topic**: Data analysis, subsetting data, plotting
+ - **Lesson content URL**: <https://github.com/UofTCoders/studyGroup/blob/gh-pages/lessons/pandas_walkthrough/>
+
+The lesson was uploaded as a Jupyter notebook. 
+
+If you'd like to follow along in IPython Notebook, do the following:
+
+  - Install [Jupyter Notebook](http://jupyter.readthedocs.org/en/latest/install.html)
```
Comment (QuLogic) [inline lessons/pandas_walkthrough/lesson.md]: Use https link.

**Comment by QuLogic** [comment]
Both github.io links in the notebook should use https.

The `gapminderDataFiveYear.txt` file is not included here, nor is there  link to it.

**Comment by QuLogic** [comment]
Can't comment on the notebook itself, so:
- I'm not sure if you want the sorta completed outputs to be there; there are a few errors near the end. (like, what is `test2`?) Maybe do a Cell->Run All to be sure that everything will work correctly in order.
- When using `idxmax` to find the maximum life expectancy in Asia, you used `df.iloc`, but `idxmax` returns the _index_ (i.e., row label), not the row number. So, you technically need to use `df.loc` (though it makes no difference here since the index is the row number when read as a table by default.) This is more evident if you `df.set_index` to a new column that has strings in it, or if you delete a few rows...
- The docstring for the `ranker` function doesn't really match what it's doing.

**Comment by derekhoward** [comment]
Thanks for the suggestions Elliott!

**Comment by QuLogic** [comment]
Funnily enough, since you're looking at gapminder, this is _really_ similar to what @edickie and @hgibling presented yesterday for the SC R lessons.

**Comment by QuLogic** [comment]
I don't suppose you want to rebase to get rid of the add/remove/add/remove of the notebook file?

**Comment by QuLogic** [comment]
I squashed and merged this on the command-line. I also added a link from the lesson to the notebook in nbviewer so people who don't have Jupyter installed can see the rendered version.

**Pull request closed without merging**